evm_vs_ewasm_opcode_table.md

```diff
@@ -0,0 +1,120 @@
+0x42: TIMESTAMP       |      getBlockTimestamp
+0x43: NUMBER          |      getBlockNumber
+0x44: DIFFICULTY      |      getBlockDifficulty
+0x45: GASLIMIT        |      getBlockGasLimit
+0x54: SLOAD           |      storageLoad
+0x55: SSTORE          |      storageStore
+0x58: PC              |      -
+0x5a: GAS             |      getGasLeft
+0xa0: LOG0            |      log(0,..)
+...                   |
+0xa4: LOG4            |      log(4,..)
+0xf0: CREATE          |      create
+0xf1: CALL            |      call
+0xf2: CALLCODE        |      callCode
+0xf3: RETURN          |      return
+0xf4: DELEGATECALL    |      callDelegate
+0xfa: STATICCALL      |      callStatic
+0xfd: REVERT          |      revert
+0xff: SELFDESTRUCT    |      selfDestruct
+
+
+
+## EVM opcodes vs WebAssembly instructions
+
+Whereas EVM is an untyped VM with a 256-bit word size, WebAssembly is a typed VM with 32-bit and 64-bit word sizes. The two wasm base types are integers and floats, which when combined with the two word sizes, create four types: i32, i64, f32, f64.
+
+Ethereum WebAssembly, or ewasm, is a subset of wasm. Only the integer types (i32 and i64) are supported, floating point types and floating point instructions are not supported. In total, 60 wasm instructions are supported in ewasm:
+* 10 control flow instructions
+* 11 basic instructions
+* 19 integer arithmetic instructions
+* 10 integer comparison instructions
+* 3 integer conversion instructions
+* 7 memory instructions (load, store, extend)
+
+ewasm excludes the wasm floating point instructions:
+* 14 floating point arithmetic instructions
+* 6 floating point comparison instructions
+* 7 floating point conversion instructions
+
+To reference wasm instructions, see the [wasm spec](https://github.com/WebAssembly/design/blob/master/Semantics.md) or [reference manual](https://github.com/sunfishcode/wasm-reference-manual/blob/master/WebAssembly.md#instructions).
+
+
+### EVM opcodes that translate to wasm instructions
+
+EVM opcode            |    wasm instruction (i32/i64)
+----------------------|----------------------
+0x00: STOP            |       end
+0x01: ADD             |       i32.add
+0x02: MUL             |       i32.mul
+0x03: SUB             |       i32.sub
+0x04: DIV             |       i32.div_u
+0x05: SDIV            |       i32.div_s
+0x06: MOD             |       i32.rem_u
+0x07: SMOD            |       i32.rem_s
+0x08: ADDMOD          |       -
+0x09: MULMOD          |       -
+0x0a: EXP             |       -
+0x0b: SIGNEXTEND      |       -
+0x10: LT              |       i32.lt_u
+0x11: GT              |       i32.gt_u
+0x12: SLT             |       i32.lt_s
+0x13: SGT             |       i32.gt_s
+0x14: EQ              |       i32.eq
+0x15: ISZERO          |       i32.eqz
+0x16: AND             |       i32.and
+0x17: OR              |       i32.or
+0x18: XOR             |       i32.xor
+0x19: NOT             |       -
+0x1a: BYTE            |       - [i32.load8]
```

Comment:
Should this say [i32.load8]?

Reply:
Its in brackets because I wasn't sure about this pair, but I'll remove them.